test_util/test_launch.py

```diff
@@ -0,0 +1,156 @@
+import json
+
+import pytest
+import yaml
+from docopt import DocoptExit
+
+import test_util
+from pkgpanda.util import load_json
+from test_util.helpers import Host
+from test_util.launch import get_launcher, main
+
+
+def check_cli(cmd):
+    assert main(cmd.split(' ')) == 0, 'CLI args failed! {}'.format(cmd)
+
+
+def check_success(capsys, tmpdir, config_str):
+    """
+    Runs through the required functions of a laucnher and then
+    runs through the default usage of the script for a
+    given config path and info path, ensuring each step passes
+    if all steps finished successfully, this parses and returns the generated
+    info JSON and stdout description JSON for more specific checks
+    """
+    # Test launcher directly first
+    config = yaml.load(config_str)
```

Comment:
nit: may want to use `safe_load` to prevent doing things users can't / make sure we're parsing yaml in the same way user yaml will be parsed then.